google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerSchemaManagementTool.java

```diff
@@ -34,6 +45,111 @@
+   * cause an error to be returned for the migration.
+   */
+  static class SpannerDdlTransactionIsolator implements DdlTransactionIsolator {
+    private static final AbstractStatementParser PARSER =
+        AbstractStatementParser.getInstance(Dialect.GOOGLE_STANDARD_SQL);
+    private final Method createStatementMethod;
+    private final Method executeMethod;
+    private final DdlTransactionIsolator delegate;
+
+    SpannerDdlTransactionIsolator(DdlTransactionIsolator delegate) throws NoSuchMethodException {
+      this.delegate = delegate;
+      this.createStatementMethod = Connection.class.getDeclaredMethod("createStatement");
+      this.executeMethod = Statement.class.getDeclaredMethod("execute", String.class);
+    }
+
+    @Override
+    public JdbcContext getJdbcContext() {
+      return delegate.getJdbcContext();
+    }
+
+    @Override
+    public void prepare() {
+      delegate.prepare();
+    }
+
+    @Override
+    public Connection getIsolatedConnection() {
+      Connection delegateConnection = this.delegate.getIsolatedConnection();
+      // Create a proxy for the connection that will override the call to
+      // Connection#createStatement().
+      return (Connection)
+          Proxy.newProxyInstance(
+              delegateConnection.getClass().getClassLoader(),
+              new Class[] {Connection.class},
+              (proxy, method, args) -> {
+                // Only handle the Connection#createStatement() differently.
+                // All other methods are just passed through.
+                if (method.equals(createStatementMethod)) {
+                  // Create a proxy for the returned Statement that will override the behavior of
+                  // Statement#execute(String).
+                  Statement delegateStatement = delegateConnection.createStatement();
+                  return Proxy.newProxyInstance(
+                      delegateConnection.getClass().getClassLoader(),
+                      new Class[] {Statement.class},
+                      (proxy1, method1, args1) -> {
+                        // Only handle the Statement#execute(String) method differently.
+                        // All other methods are just passed through.
+                        if (method1.equals(executeMethod)
+                            && args1 != null
+                            && args1.length == 1
+                            && args1[0] instanceof String) {
+                          // Check if the statement that is being executed is either `START BATCH
+                          // DDL` or `RUN BATCH`.
+                          String sql = (String) args1[0];
+                          ParsedStatement statement =
+                              PARSER.parse(com.google.cloud.spanner.Statement.of(sql));
+                          if (statement.getType() == StatementType.CLIENT_SIDE) {
+                            if (statement.getClientSideStatementType()
```

Comment:
Could you combine two `if` statements in line 114-115?

Reply:
Done